packages/mongodb-chatbot-ui/src/services/conversations.ts

```diff
@@ -165,12 +163,14 @@ export type AddMessageRequestBody = {
 
 export type ConversationServiceConfig = {
   serverUrl: string;
-  fetchOptions?: ConversationFetchOptions;
+  fetchOptions?: ConversationFetchOptions | (() => ConversationFetchOptions);
```

Comment:
dont love the overloading of this w/ 2 types as a design principle but meh

Reply:
Yeah maybe not ideal but I've seen it before. Mainly here for backwards compatibility just in case.